### ISSUE TITLE
Training dataset format

### ISSUE THREAD

**Issue by fgraffitti-cyberhawk** [ultralytics/ultralytics#3964]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Feeling a bit dumb here but I can't really find anywhere what is the format to train yolo8 models with custom datasets.
I'm interested in trying both object detection and instance segmentation, I have datasets in COCO format (json file) but I can easily convert them to other formats. However, the only thing I can find in the repo regarding training is the following line of code:
`model.train(data='coco128.yaml', epochs=100, imgsz=640)`
that really doesn't tell me much about what's the format for the data, the annotations, the yaml file etc.
Any help (or link to resources that I might have missed) is welcome!

**Comment by AnnaFHub** [comment]
On my search for information on how to structure the folders of my custom dataset, I came across this issue. I now am a bit confused as the resources linked in the initial post specify two different folder structures. Namely the order of train/val and images/labels folders are switched.

Now I was wondering if there is a general preference for one of the both formats?

[https://docs.ultralytics.com/datasets/#contribute-new-datasets](https://docs.ultralytics.com/datasets/#contribute-new-datasets):
![image](https://github.com/user-attachments/assets/115c3fc4-d982-4c64-8c70-bc4627af72f1)

vs

[https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format](https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format)
![image](https://github.com/user-attachments/assets/4fad7b0c-c3c7-4114-99c6-259952583a07)

Also I have seen elsewhere that the images and label files are put into one folder together, e.g.:
- dataset
    - train
        - img1.jpg
        - img1.txt
        - img2.jpg
        - img2.txt
    - val
        - img3.jpg
        - img3.txt
        
Would that also work?

**Comment by glenn-jocher** [comment]
@AnnaFHub thank you for your question! For training YOLO models, the standard and recommended structure is to use **separate directories** for images and labels under `train`, `val`, and optionally `test` folders. For example:

```
dataset/
├── images/
│   ├── train/
│   ├── val/
├── labels/
│   ├── train/
│   ├── val/
```

This format is consistent with the **Ultralytics YOLO dataset format** ([reference](https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format)) and offers better organization and scalability. The example where images and labels are combined in one folder is not supported and may cause issues.

If converting datasets, ensure they follow the recommended format above for seamless training. Let us know if you encounter further confusion!
